### PR TITLE
NPR rename, better Android SDK detection, more podcasting user agents.

### PR DESCRIPTION
**Pull request by mindreader**
### Description:

This is a remake of this PR: https://github.com/matomo-org/device-detector/pull/7566 Before anyone could think about merging it, it got a pretty bad conflict with some other change that got merged. Due to the difficulty of the android fixes I just ran out of steam and could not fix it in a timely manner, so here I'm starting from scratch.

More things from the opawg2 podcast useragent list. Major things:

Big thing: Android SDK detection is a lot better, as someone requested in feedback on the other pr, and I regret taking it on but it is done now.

Rename NPR One -> NPR
Youtube Music detection
Some Amazon Echo detection

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: regexes/client/mobile_apps.yml [hunk 1]
```diff
@@ -1908,7 +1909,7 @@
   version: '$1'
 
 # The free NRC app offers you the latest news and the digital version of the paper newspaper.
-- regex: '(?:NRC-Nieuws/|nl\.nrc\.nrcapp )([\d\.]+)'
+- regex: '(?:NRC-Nieuws/|nl.nrc.nrcapp |com.twipemobile.nrc )([\d\.]+)'
```
Comment (sanchezzzhak) [inline regexes/client/mobile_apps.yml:1912]: regex: '(?:NRC-Nieuws/|nl\.nrc\.nrcapp |com\.twipemobile\.nrc )([\d\.]+)'

we just recently added escaping for any chars (I wouldn't want anyone to do double work again)

File: regexes/bots.yml [hunk 2]
```diff
@@ -3807,6 +3807,16 @@
   category: 'Crawler'
   url: 'https://github.com/aaamoon/copilot-gpt4-service'
 
+- regex: '^pdrl.fm'
```
Comment (sanchezzzhak) [inline regexes/bots.yml:3810]:
```
- regex: '^pdrl\.fm'
```

File: regexes/client/mobile_apps.yml [hunk 3]
```diff
@@ -215,6 +215,11 @@
   name: 'Yo WhatsApp'
   version: '$1'
 
+# https://music.youtube.com
+- regex: '(?:^com.google.android.apps.youtube.music/|^YouTubeMusic(?:Dev)?/)([\d\.]+)?'
```
Comment (sanchezzzhak) [inline regexes/client/mobile_apps.yml:219]:
```
- regex: '(?:^com\.google\.android\.apps\.youtube\.music/|^YouTubeMusic(?:Dev)?/)([\d\.]+)?'
```

File: regexes/client/mobile_apps.yml [hunk 4]
```diff
@@ -2246,6 +2247,30 @@
   name: 'IPTV Pro'
   version: '$1'
 
+- regex: '^com.audials(?:\.paid)?/([\d\.]+)'
```
Comment (sanchezzzhak) [inline regexes/client/mobile_apps.yml:2250]:
```
- regex: '^com\.audials(?:\.paid)?/([\d\.]+)'
```

File: regexes/client/mobile_apps.yml [hunk 5]
```diff
@@ -2246,6 +2247,30 @@
...
+
+- regex: '^Metacast/([\d\.]+)'
+  name: 'Metacast'
+  version: '$1'
+
+- regex: '^mowPod/([\d\.]+)'
+  name: 'mowPod'
+  version: '$1'
+
+- regex: 'com.meecel.feedreader.RssDemonAd'
```
Comment (sanchezzzhak) [inline regexes/client/mobile_apps.yml:2266]:
```
- regex: 'com\.meecel\.feedreader\.RssDemonAd'
```

File: regexes/client/mobile_apps.yml (hunk 1 above)
Comment (mindreader) [reply regexes/client/mobile_apps.yml]: Good to know!

**Review by sanchezzzhak — APPROVED** [review]

**Pull request merged**